InputSequenceProvider.java

```diff
@@ -22,8 +22,30 @@ public InputSequenceProvider(String[] args) {
       return reader.readSequence();
     } else {
       String[] tempSequence = firstSequence;
+      /* to do : firstSequence field --> currentSequence field */ 
       firstSequence = new String[0];
       return tempSequence;
     }
   }
+  
+  /**
+   * Gets the next input triangle sequence (a sequence of triangle sides).
+   * When called for the first time, returns the value of firstSequence field as a sequence,
+   * and sets that field empty.
+   * When the subsequent callings occur, requests sequences from the console. 
+   * 
+   * @return the sequence.
+   */ 
+  public String[] provideInputTriangle() {
+    // to do : how should I do this not violating DRY ?
```

Comment:
"Do not Repeat Yourself"
this method is almost a Ctrl+C / Ctrl+V of the provideInputSequence()